src/AutoRest.CSharp/Common/Output/Builders/MultipartSerializationMethodsBuilder.cs

```diff
@@ -0,0 +1,265 @@
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using AutoRest.CSharp.Common.Input;
+using AutoRest.CSharp.Common.Output.Expressions.KnownValueExpressions;
+using AutoRest.CSharp.Common.Output.Expressions.Statements;
+using AutoRest.CSharp.Common.Output.Expressions.ValueExpressions;
+using AutoRest.CSharp.Common.Output.Models;
+using AutoRest.CSharp.Common.Output.Models.Serialization.Multipart;
+using AutoRest.CSharp.Common.Output.Models.Types;
+using AutoRest.CSharp.Generation.Types;
+using AutoRest.CSharp.Output.Models;
+using AutoRest.CSharp.Output.Models.Serialization;
+using AutoRest.CSharp.Output.Models.Shared;
+using AutoRest.CSharp.Output.Models.Types;
+using static AutoRest.CSharp.Common.Output.Models.Snippets;
+
+namespace AutoRest.CSharp.Common.Output.Builders
+{
+    internal class MultipartSerializationMethodsBuilder
+    {
+        public static IEnumerable<Method> BuildMultipartSerializationMethods(MultipartObjectSerialization multipart)
+        {
+            yield return new Method(
+                new MethodSignature(
+                    "SerializeMultipart",
+                    null,
+                    null,
+                    MethodSignatureModifiers.Private,
+                    typeof(BinaryData),
+                    null,
+                    new Parameter[] { KnownParameters.Serializations.Options }),
+                BuildMultipartSerializationMethodBody().ToArray());
+            yield return new Method(
+                new MethodSignature(
+                    Configuration.ApiTypes.ToMultipartRequestContentName,
+                    null,
+                    null,
+                    MethodSignatureModifiers.Internal | MethodSignatureModifiers.Virtual,
+                    Configuration.ApiTypes.MultipartRequestContentType,
+                    null,
+                    Array.Empty<Parameter>()),
+                BuildToMultipartRequestContentMethodBody(multipart).ToArray());
+        }
+        public static SwitchCase BuildMultipartWriteSwitchCase(MultipartObjectSerialization multipart, ModelReaderWriterOptionsExpression options)
+        {
+            return new SwitchCase(
+                Serializations.MultipartFormat,
+                Return(
+                    new InvokeInstanceMethodExpression(
+                        null,
+                        new MethodSignature(
+                            $"SerializeMultipart",
+                            null,
+                            null,
+                            MethodSignatureModifiers.Private,
+                            typeof(BinaryData),
+                            null,
+                            new[]
+                            {
+                                KnownParameters.Serializations.Options
+                            }),
+                        new ValueExpression[]
+                        {
+                            options
+                        })));
```

Comment:
if we want to invoke a method with a given parameter list, please just do:
```
new InvokeInstanceMethodExpression(null, "SerializeMultipart", new[] {options})
```
it has a ctor that just takes the method name.
The overload taking `MethodSignature` we should only use it when we have a `MethodSignature`.

Reply:
same comment applies to all the similar cases in this file.